packages/playwright-core/src/server/trace/recorder/tracing.ts

```diff
@@ -201,10 +216,16 @@ export class Tracing extends SdkObject implements InstrumentationListener, Snaps
     await this._appendTraceOperation(async () => {
       const oldNetworkFile = state.networkFile;
       state.traceName = name;
-      state.traceFile = path.join(state.tracesDir, name + '.trace');
-      state.networkFile = path.join(state.tracesDir, name + '.network');
+      state.traceFile = {
+        file: path.join(state.tracesDir, name + '.trace'),
+        buffer: [],
+      };
+      state.networkFile = {
+        file: path.join(state.tracesDir, name + '.network'),
```

Comment:
I think we should flush files here, otherwise we might be copying an uncompleted old network file at least.

Reply:
I thought we only call it in the startChunk?